diff --git a/lib/printf-args.c b/lib/printf-args.c
index 4978914..e994239 100644
--- a/lib/printf-args.c
+++ b/lib/printf-args.c
@@ -1,5 +1,6 @@
 /* Decomposed printf argument list.
-   Copyright (C) 1999, 2002-2003, 2005-2007 Free Software Foundation, Inc.
+   Copyright (C) 1999, 2002-2003, 2005-2007, 2009 Free Software
+   Foundation, Inc.

    This program is free software; you can redistribute it and/or modify
    it under the terms of the GNU General Public License as published by
@@ -89,7 +90,7 @@ PRINTF_FETCHARGS (va_list args, arguments *a)
 	   where wint_t is 'unsigned short'.  */
 	ap->a.a_wide_char =
 	  (sizeof (wint_t) < sizeof (int)
-	   ? va_arg (args, int)
+	   ? (wint_t) va_arg (args, int)
 	   : va_arg (args, wint_t));
 	break;
 #endif
