diff --git a/lib/regex_internal.c b/lib/regex_internal.c
index 7e8c1bc..e68c9da 100644
--- a/lib/regex_internal.c
+++ b/lib/regex_internal.c
@@ -16,6 +16,8 @@
    You should have received a copy of the GNU General Public License along
    with this program; if not, see <http://www.gnu.org/licenses/>.  */

+#include "verify.h"
+#include "intprops.h"
 static void re_string_construct_common (const char *str, Idx len,
 					re_string_t *pstr,
 					RE_TRANSLATE_TYPE trans, bool icase,
@@ -1393,7 +1395,10 @@ static void
 internal_function
 re_node_set_remove_at (re_node_set *set, Idx idx)
 {
-  if (idx < 0 || idx >= set->nelem)
+  verify (! TYPE_SIGNED (Idx));
+  /* if (idx < 0)
+     return; */
+  if (idx >= set->nelem)
     return;
   --set->nelem;
   for (; idx < set->nelem; idx++)
