diff --git a/lib/regex_internal.h b/lib/regex_internal.h
index 439444c..7242084 100644
--- a/lib/regex_internal.h
+++ b/lib/regex_internal.h
@@ -827,7 +827,8 @@ re_string_wchar_at (const re_string_t *pstr, Idx idx)
 # ifndef NOT_IN_libc
 static int
 internal_function __attribute__ ((pure, unused))
-re_string_elem_size_at (const re_string_t *pstr, Idx idx)
+re_string_elem_size_at (const re_string_t *pstr _UNUSED_PARAMETER_,
+			Idx idx _UNUSED_PARAMETER_)
 {
 #  ifdef _LIBC
   const unsigned char *p, *extra;
