--- man/help2man-orig
+++ man/help2man	2013-09-10 07:31:28.352365301 +0200
@@ -412,6 +412,7 @@
 my $PAT_FILES         = _('Files');
 my $PAT_EXAMPLES      = _('Examples');
 my $PAT_FREE_SOFTWARE = _('This +is +free +software');
+my $PAT_INFO          = _('For +complete +documentation');
 
 # Start a new paragraph (if required) for these.
 s/([^\n])\n($PAT_BUGS|$PAT_AUTHOR) /$1\n\n$2 /og;
@@ -444,6 +445,12 @@
 	next;
     }
 
+    # Skip any texinfo reference as that's handled separately
+    if (s/($PAT_INFO).*\n//o)
+    {
+        next;
+    }
+
     # Copyright section
     if (/^Copyright /)
     {
@@ -622,7 +629,7 @@
 .B %s
 programs are properly installed at your site, the command
 .IP
-.B info %s
+.B info coreutils \(aq%s invocation\(aq
 .PP
 should give you access to the complete manual.
 EOT
