diff --git a/modules/tempname b/modules/tempname
index 7fafd72..4703517 100644
--- a/modules/tempname
+++ b/modules/tempname
@@ -1,5 +1,5 @@
 Description:
-gen_tempname() function: create a private temporary file or directory.
+gen_tempname, gen_tempname_len: create a private temporary file or directory.

 Files:
 lib/tempname.c
@@ -11,7 +11,9 @@ extensions
 fcntl-h
 gettimeofday
 lstat
+randint
 secure_getenv
+stdbool
 stdint
 sys_stat
 sys_time
