diff --git a/top/maint.mk b/top/maint.mk
index ad60c8fb11..3475f3733c 100644
--- a/top/maint.mk
+++ b/top/maint.mk
@@ -1740,6 +1740,7 @@ web-manual:
 	$(AM_V_GEN)test -z "$(manual_title)" \
 	  && { echo define manual_title in cfg.mk 1>&2; exit 1; } || :
 	$(AM_V_at)cd '$(srcdir)/doc'; \
+	  MAKEINFO="$(MAKEINFO)" \
 	  $(SHELL) ../$(_build-aux)/gendocs.sh $(gendocs_options_) \
 	     -o '$(abs_builddir)/doc/manual' \
 	     --email $(PACKAGE_BUGREPORT) $(PACKAGE) \
